### ISSUE TITLE
[4.1 Regression] More touch issues after widget interaction

### ISSUE THREAD

**Issue by janusw** [Mapsui/Mapsui#2262]
**Mapsui Version**
4.1.1

**Mapsui Platform**
MAUI / Xamarin

**Device**
iOS 17

**Describe the bug**
Follow-up to #2197. After interacting with a widget in a specific way (see below), map panning is broken.

**To Reproduce**
1. Run Mapsui.Samples.Maui.MapView on an iOS device (iPhone/iPad).
2. Select a sample, let's say Demo -> OpenStreetMap.
3. Touch the map with one finger, drag onto one of the widgets (zoom in/out or center) and lift off the finger.
4. -> The widget is triggered and map panning is broken now: single-finger dragging zooms now.
5. Repeat step 3.
6. -> Now map panning is broken even more: single-finger dragging has no effect any more.

**Expected behavior**
Single-finger dragging should continue to work as before (panning the map). 4.0.0 works well, 4.1.1 fails.

**Additional context**
The issue is similar to #2197, and therefore probably related to the changes in PR #2133, #2190 and #2203.

**Comment by janusw** [comment]
>     3. Touch the map with one finger, drag onto one of the widgets (zoom in/out or center) and lift off the finger.
> 
>     4. -> The widget is triggered and map panning is broken now: single-finger dragging zooms now.

Note: With 4.0.0, the gesture in step 3 does not trigger the widget.

**Comment by charlenni** [comment]
@janusw What do you expect in this case? Is it, when you enter the widget in pan mode, you still want to pan? So if you enter the widget with an already touched state, you like no event for the widget.

I assume, that this should be checked [here](https://github.com/Mapsui/Mapsui/blob/83d393f2a2b8bd6aee2cf5169f4ea4ffd43e84ef/Mapsui.UI.Shared/MapControl.cs#L653). There should be a distinguishing flag, if the widget likes to have all events (click and move) or only that events, that corresponds to clicks.

When I look through the code, we are not very clear with the naming. What is a click, a touch, a move or a pan? Sometimes we use the mobile version, sometimes the desktop version. So is a touchable widget on, that could get only clicks or clicks and pans? And if I'm understanding right, then @inforithmics introduced this because of the edit functionality. Is this something belonging to widgets? For me, widgets are always overlays, a layer fixed on the screen coordinates on top of the map. But in this case the definition seems not to be right, correct?

**Comment by janusw** [comment]
> @janusw What do you expect in this case? Is it, when you enter the widget in pan mode, you still want to pan? So if you enter the widget with an already touched state, you like no event for the widget.

To be very clear here: The reproduction procedure described above is not something like an actual use case for me, it's more some obscurity that I came across by accident.

I don't have a very strong preference on the question whether the widget should be triggered in this situation or not (but would slightly lean toward the old behavior of not triggering the widget if the initial tap occurred outside of the widget).

The main problem here is that the map navigation (panning and zooming) is completely broken afterwards (probably because the initial touch point is never removed).

**Comment by pauldendulk** [comment]
I think the problem is that to determine if a widget is touched you need to know the start and end location. Both should be within the widget.

**Comment by pauldendulk** [comment]
oh GetTouchedWidget actually has a separate argument for startScreenPosition
```csharp
public static IEnumerable<IWidget> GetTouchedWidget(MPoint screenPosition, MPoint startScreenPosition,
        IEnumerable<IWidget> widgets)
```
but is called with the current position twice
```csharp
var touchedWidgets = WidgetTouch.GetTouchedWidget(position, position, touchableWidgets);
```

**Comment by pauldendulk** [comment]
> When I look through the code, we are not very clear with the naming. What is a click, a touch, a move or a pan? Sometimes we use the mobile version, sometimes the desktop version. So is a touchable widget on, that could get only clicks or clicks and pans? 

This is something to improve in v5. We should try to channel the platform specific logic into shared components early on. Both touch tapping and mouse clicking could be treaded at 'pointer' actions. So, we would have a pointerDown, pointerMove, pointerUp event. When pinch is used this not a pointer event, but in some cases there might not be an easy way to distinguish between the two. For instance a touch can start with a single finger down and then become a pinch. We have to think about a clear solution for such cases.

**Comment by janusw** [comment]
With release `4.1.2-preview.1`, I could still see a variation of this bug. With `4.1.2-preview.2`, I no longer see any problem. I assume it was fixed via #2326 (thanks, Paul!).

**Comment by janusw** [comment]
Tried on iOS and Android with Xamarin.Forms and Maui. All fine.
